### PR TITLE
Enable cancelling active listeners when unsubscribing

### PR DESCRIPTION
**Pull request by markerikson**
This PR:

- Renames the `removeAllListeners` action creator to `clearAllListeners`
- Updates the `unsubscribe` callbacks to optionally accept `{cancelActive: true}` to cancel active listeners
- Updates the API doc to be more clear about cancellation behavior while unsubscribing
- Updates the last couple `@alpha` type labels to be `@public`

### CODE REVIEW

File: packages/toolkit/src/listenerMiddleware/tests/listenerMiddleware.test.ts [hunk 1]
```diff
@@ -478,6 +478,80 @@ describe('createListenerMiddleware', () => {
       expect(effect.mock.calls).toEqual([[testAction1('a'), middlewareApi]])
     })
 
+    test.only('can cancel an active listener when unsubscribing directly', async () => {
```
Comment (FaberVitale) [inline packages/toolkit/src/listenerMiddleware/tests/listenerMiddleware.test.ts:481]: `.only` should be removed.

**Review by FaberVitale — CHANGES_REQUESTED** [review]

File: packages/toolkit/src/listenerMiddleware/tests/listenerMiddleware.test.ts (hunk 1 above)
Comment (markerikson) [reply packages/toolkit/src/listenerMiddleware/tests/listenerMiddleware.test.ts:481]: DRAT thank you!

File: packages/toolkit/src/listenerMiddleware/index.ts [hunk 2]
```diff
@@ -14,14 +14,15 @@ import type {
...
-  Unsubscribe,
+  UnsubscribeListener,
   TakePattern,
   ListenerErrorInfo,
   ForkedTaskExecutor,
   ForkedTask,
   TypedRemoveListener,
   TaskResult,
   AbortSignalWithReason,
+  UnsubscribeOptions,
```
Comment (FaberVitale) [inline packages/toolkit/src/listenerMiddleware/index.ts:25]:
```suggestion
  UnsubscribeListenerOptions,
```

Comment (FaberVitale) [reply packages/toolkit/src/listenerMiddleware/index.ts:25]: 2 Suggestions:

1. Rename `UnsubscribeOptions` -> `UnsubscribeListenerOptions`
2. export `UnsubscribeListenerOptions`

Comment (markerikson) [reply packages/toolkit/src/listenerMiddleware/index.ts]: Good idea! force-pushed

File: packages/toolkit/src/listenerMiddleware/index.ts [hunk 3]
```diff
@@ -279,8 +281,16 @@ const defaultErrorHandler: ListenerErrorHandler = (...args: unknown[]) => {
   console.error(`${alm}/error`, ...args)
 }
 
+function cancelActiveListeners(
```
Comment (FaberVitale) [inline packages/toolkit/src/listenerMiddleware/index.ts:284]:
```suggestion
const cancelActiveListeners = (
```

Comment (FaberVitale) [reply packages/toolkit/src/listenerMiddleware/index.ts]: Definitely a minor change but we save few bytes. 

Note that you also need to hoist this function above `createClearListenerMiddleware` if you want to accept this suggestion.

**Comment by markerikson** [comment]
Awright. Merging, and releasing shortly!

**Comment by FaberVitale** [comment]
> Awright. Merging, and releasing shortly!

 👍

**Pull request merged**